src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsManager.cs

```diff
@@ -12,14 +11,14 @@ namespace System.Data.SqlClient.SNI
     /// </summary>
     internal class SNIMarsManager
     {
-        public static readonly SNIMarsManager Singleton = new SNIMarsManager();
-        private ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections = new ConcurrentDictionary<SNIHandle, SNIMarsConnection>();
+        private readonly ConcurrentDictionary<SNIHandle, SNIMarsConnection> _connections;
 
         /// <summary>
         /// Constructor
         /// </summary>
         public SNIMarsManager()
```

Comment:
The SNIMarsManager needs to shared between a physical connection state object and the MARS session state objects that are wrapped around that physical connection. The SNIMarsManager gets copied to subsequent state objects in CreateSessionHandle. By making this an instanced class rather than static, the mars manager will actually be garbage collected once all the owning state objects are disposed.